From a5ca90c2ff73729371e99662f12f7f19be311e60 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sun, 28 Jun 2020 22:47:30 +0200
Subject: [PATCH 32/49] ARM: dts: stm32: Add AV96 DTO for DH 644-100 DSI
 mezzanine with Orisetech OTM8009A display

Add DT overlay for the AV96 board for DH 644-100 mezzanine card with
Orisetech OTM8009A DSI display attached on top.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |  1 +
 ...-avenger96-overlay-644-100-x6-otm8009a.dts | 61 +++++++++++++++++++
 2 files changed, 62 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-644-100-x6-otm8009a.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b940ea6fdf677..95c4817056ef2 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1218,6 +1218,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp153c-dhcor-drc-compact.dtb \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
+	stm32mp15xx-avenger96-overlay-644-100-x6-otm8009a.dtbo \
 	stm32mp15xx-avenger96-overlay-fdcan1-x6.dtbo \
 	stm32mp15xx-avenger96-overlay-fdcan2-x6.dtbo \
 	stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dtbo \
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-644-100-x6-otm8009a.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-644-100-x6-otm8009a.dts
new file mode 100644
index 0000000000000..87666788bda04
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-644-100-x6-otm8009a.dts
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+#include <dt-bindings/gpio/gpio.h>
+
+/dts-v1/;
+/plugin/;
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	phy-dsi-supply = <&reg18>;
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			dsi_in: endpoint {
+				remote-endpoint = <&ltdc_ep1_out>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			dsi_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
+
+	panel@0 {
+		compatible = "orisetech,otm8009a";
+		reg = <0>;
+		reset-gpios = <&gpiod 8 GPIO_ACTIVE_LOW>;
+		power-supply = <&v3v3>;
+		status = "okay";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+	};
+};
+
+&ltdc {
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ltdc_ep1_out: endpoint@1 {
+			reg = <1>;
+			remote-endpoint = <&dsi_in>;
+		};
+	};
+};
-- 
2.39.2

